### ISSUE TITLE
Utility class containing `hasOwnProperty` static utility method is incorrectly classified as target object in no-prototype-builtins rule

### ISSUE THREAD

**Issue by kafkas** [eslint/eslint#13961]
**What rule do you want to change?**
`no-prototype-builtins`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
This should be the default behaviour

**Please provide some example code that this change will affect:**

Consider the following class that contains static utility methods.
```js
export class ObjectUtils {
  // ...
  static hasOwnProperty(object, property) {
    return Object.prototype.hasOwnProperty.call(object, property);
  }
}
```
The following call results in an Eslint error
```js
ObjectUtils.hasOwnProperty(obj, property)
// Do not access Object.prototype method 'hasOwnProperty' from target object. eslint(no-prototype-builtins)
```

**Comment by aladdin-add** [comment]
this is somehow the limitation of static linting tool, as eslint didn't know the type of `ObjectUtils`.

**Comment by kafkas** [comment]
Thanks, already done that. Yeah, it's not a huge issue but I just thought it wouldn't be too difficult for Eslint to check whether ObjectUtils refers to a class or an instance.
